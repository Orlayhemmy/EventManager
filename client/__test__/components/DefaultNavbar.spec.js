import React from 'react';
import DefaultNavbar from '../../components/Navbar/Template/Content/Index';

describe('Default navbar', () => {
  const props = {
    auth: {
      user: { isAdmin: true }
    }
  };
  const wrapper = shallow(<DefaultNavbar navbarProps={props} />);
  it('should render the component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should switch the default navbar when user is authenticated', () => {
    wrapper.setProps({
      navbarProps: {
        auth: {
          user: { isAdmin: false }
        }
      }
    });
    expect(wrapper.find('#usernavbar').length).toEqual(1);
  });
});
