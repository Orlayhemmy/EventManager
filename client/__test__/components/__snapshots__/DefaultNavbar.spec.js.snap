// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default navbar should render the component based 1`] = `
<AuthNavbar
  id="authnavbar"
  navbarProps={
    Object {
      "auth": Object {
        "user": Object {
          "isAdmin": true,
        },
      },
    }
  }
/>
`;
